### PR TITLE
Add new data type to be serialized after response into json.

### LINKED ISSUE #549: Add ability to return a model which returns a json response
Currently when returning a json response, we have to serialize the model first like this:

```python
def show(self):
    return User.where('key', 'value').get().serialize()
```

It seems sort of redundent to always have to serialize so we should be able to just return the model and Masonite should serialize for us:

```python
def show(self):
    return User.where('key', 'value').get()
```

This just needs a new if statement inside the `ResponseMiddleware` inside `core`

### PR DESCRIPTION
**Pull request by Marlysson**
@josephmancuso @vaibhavmule 
Closes #549 

Starting implementing the conditionals to verify each Response's type.

Steps that I consider to getting this done.

**IMPLEMENTATION**
- [ ] Verify each type `Model`/`Collection`

**TEST**

- [ ] Create UserMock simulating an Orator model
- [ ] Create controller test to get the method's return and verify his type.
- [ ] Verify wheter return is a json.

### CODE REVIEW

File: masonite/response.py [hunk 1]
```diff
@@ -87,6 +89,10 @@ def view(self, view, status=200):
 
         if isinstance(view, dict) or isinstance(view, list):
             return self.json(view)
+        else if isinstance(view, Collection):
+            return view.serialize()
+        else if isinstance(view, Model):
+            return view.to_json()
```
Comment (josephmancuso) [inline masonite/response.py]: it's `elif` not `else if` 😉

Comment (Marlysson) [reply masonite/response.py]: It's Java mind here.. So sorry.. 😃

Comment (josephmancuso) [reply masonite/response.py]: haha 😅

Comment (vaibhavmule) [reply masonite/response.py]: Line number 95. It should be `return self.json(view.to_dict())` as you want add JSON headers while returning response

**Comment by josephmancuso** [comment]
idk why i thought this was in the `ResponseMiddleware` 🤔 . Must have forgotten we moved it out and put it into a response class.

**Comment by josephmancuso** [comment]
as for a test, it would likely look something like:

```python
    def test_view_serialize_collection(self):

        self.response.view(MockUser().all())
        assert self.app.make('Response') == "{'1', '2'}"
```

or whatever that json response is of `Collection([1,2])` is

**Comment by Marlysson** [comment]
Don't necessary instantiate a Controller and get the `Response` returned by some method ? 

Lol, easier than I thought..

File: masonite/response.py (hunk 1 above)
Comment (vaibhavmule) [inline masonite/response.py]: You would want to return json with json headers. this line should be. 

```suggestion
            return self.json(view.to_dict())
```

Comment (josephmancuso) [reply masonite/response.py]: oh good catch

Comment (Marlysson) [reply masonite/response.py]: The View has the `to_dict` method?

Comment (josephmancuso) [reply masonite/response.py]: yes it should be to_dict() since the json method needs a dictionary

File: masonite/response.py [hunk 2]
```diff
@@ -87,6 +89,10 @@ def view(self, view, status=200):
 
         if isinstance(view, dict) or isinstance(view, list):
             return self.json(view)
+        elif isinstance(view, Collection):
+            view = view.serialize()
```
Comment (josephmancuso) [inline masonite/response.py]: looking at this again, this line should also be JSON `view = self.json(view.serialize())`

File: tests/test_response.py [hunk 3]
```diff
@@ -54,3 +67,13 @@ def test_view_can_set_own_status_code(self):
...
+
+    def test_view_should_return_a_json_response_when_retrieve_a_user_from_model(self):
+        
+        assert isinstance(MockUser(), Model)
+
+        self.response.view(MockUser().find(1))
+
+        json_response = [{'name':'TestUser', 'email': 'user@email.com' }]
+
+        assert self.app.make('Response') == json_response
```
Comment (josephmancuso) [inline tests/test_response.py]: this test is good but not correct. 

```python
json_response = "[{'name':'TestUser', 'email': 'user@email.com' }]"
assert self.app.make('Response') == json_response
```

that should be the correct assertion.

Comment (josephmancuso) [reply tests/test_response.py]: since it needs to be a string when going to the browser

Comment (josephmancuso) [reply tests/test_response.py]: That change in the response class above should fix that.

**Comment by josephmancuso** [comment]
pulled this in, made a few minor tweaks and merged into master. 🎊 🎉 . Thanks for contributing!

89aa00f5e8c236e89c6578a25ca310ad026f5586

**Pull request closed without merging**